libs/ha/ha_common_libs_k8s.py

```diff
@@ -1581,3 +1583,114 @@ def failover_pod(pod_obj, pod_list, failover_node):
                 LOGGER.error("Failed to failover pod %s to %s due to error: %s", pod,
                              failover_node, error)
                 return False, error
+
+    def mark_resource_failure(self, mnode_obj, pod_list: list, go_random: bool = True,
+                              rsc_opt: str = "mark_node_failure", rsc: str = "node",
+                              validate_set: bool = True):
+        """
+        Helper function to set resource status to Failed random if go_random.
+        :param pod_list: List of resource to be marked as failed
+        :param go_random: If True, send mark failure signal to resource randomly
+        :param mnode_obj: Master node object to fetch the resource ID
+        :param rsc_opt: Operation to be performed on resource (eg. mark_node_failure)
+        :param rsc: resource type (eg. node, cluster)
+        :param validate_set: If set to TRUE, its validate SET failure for resource
+        :return: bool, response
+        """
+        if rsc == 'node':
+            pod_info = {}
+            pod_data = {'id': None, 'status': 'offline'}
+            for pod in pod_list:
+                pod_info[pod] = pod_data.copy()
+                pod_info[pod]['id'] = mnode_obj.get_machine_id_for_pod(pod)
+
+            for pod in pod_list:
+                set_fail = self.system_random.choice([True, False]) if go_random else True
+                if set_fail:
+                    LOGGER.info('Marking %s pod as failed', pod)
+                    data_val = {"operation": rsc_opt,
+                                "arguments": {"id": f"{pod_info[pod]['id']}"}}
+                    resp = self.system_health.set_resource_signal(req_body=data_val)
+                    if not resp[0]:
+                        return False, pod_info, f"Failed to set failure status for {pod}"
+                    pod_info[pod]['status'] = 'failed'
+                    time.sleep(HA_CFG["common_params"]["30sec_delay"])
+            if validate_set:
+                return self.get_validate_resource_status(rsc_info=pod_info)
+            return True, pod_info, f"{pod_list} marked as failed"
+        return False, f"Mark failure for {rsc} is not supported yet"
+
+    def get_validate_resource_status(self, rsc_info, exp_sts=None,
+                                     mnode_obj=None, rsc: str = "node"):
+        """
+        Helper function to get and validate resource status
+        :param exp_sts: Expected status of resource.
+        :param rsc_info: Required resource to get its status,
+        dict with {pod1:{'id':, 'status':},..} or list of pods
+        :param rsc: resource type (eg. node, cluster)
+        :param mnode_obj: Master node object to fetch the resource ID
+        :return: bool, response
+        """
+        if rsc == "node":
+            if not isinstance(rsc_info, dict):
+                # Get the node ID and set expected status if only list of pods is passed
```

Comment:
Add information using logger

Reply:
> Add information using logger

This comment is just to make code understandable who wants to use this function. Added one to fetch node ID